fdbserver/storageserver.actor.cpp

```diff
@@ -8892,6 +8893,66 @@ ACTOR Future<Void> tryGetRangeForBulkLoad(PromiseStream<RangeResult> results,
 	}
 }
 
+// Utility function to process sample files during bulk load
+ACTOR static Future<Void> processSampleFiles(StorageServer* data,
+                                             std::string bulkLoadLocalDir,
+                                             std::shared_ptr<BulkLoadFileSetKeyMap> localFileSets) {
+	state BulkLoadFileSetKeyMap::const_iterator iter = localFileSets->begin();
+	state BulkLoadFileSetKeyMap::const_iterator end = localFileSets->end();
+
+	while (iter != end) {
+		const auto& [range, fileSet] = *iter;
+		if (fileSet.hasByteSampleFile()) {
+			state std::string sampleFilePath = fileSet.getBytesSampleFileFullPath();
+			state int retryCount = 0;
+			state int maxRetries = 10;
+			state Error lastError;
+
+			while (retryCount < maxRetries) {
+				try {
+					// First read all samples into memory
+					state std::vector<KeyValueRef> samples;
+					state std::unique_ptr<IRocksDBSstFileReader> reader = newRocksDBSstFileReader();
+					reader->open(abspath(sampleFilePath));
+
+					while (reader->hasNext()) {
+						KeyValue kv = reader->next();
+						samples.emplace_back(kv.key, kv.value);
+					}
+
+					// Now apply all samples in a batch
+					for (const auto& kv : samples) {
+						data->byteSampleApplySet(kv, invalidVersion);
```

Comment:
You may want to check the prefix here.

Reply:
`byteSampleApplySet` this is doing sampling given the input kv. Here, we want to do:
```data->metrics.byteSample.sample.insert(key, size);
data->metrics.notifyBytes(key, size);
data->addMutationToMutationLogOrStorage(invalidVersion, MutationRef(MutationRef::SetValue, key.withPrefix(persistByteSampleKeys.begin), kv.value));```
To ingest the byte sample metadata.